test/functional/p2p_sendtxrcncl.py

```diff
@@ -163,6 +169,11 @@ def run_test(self):
         assert not peer.sendtxrcncl_msg_received
         peer.peer_disconnect()
 
+        self.log.info("SENDTXRCNCL should not be sent if feeler")
+        peer = self.nodes[0].add_outbound_p2p_connection(P2PFeelerReceiver(), p2p_idx=2, connection_type="feeler")
```

Comment:
Shouldn't that be `p2p_idx=3` and the below bumped to `p2p_idx=4`?

Reply:
This is only used to pick a port (if none was given). The previous connection was closed, so the index can be reused.

Though, happy to change this test or remove it, since it isn't too important.

Reply:
@mzumsande I think it makes sense to just set them all to 1, because it doesn't matter, rather than handle increments.

Reply:
@MarcoFalke @naumenkogs After looking at current CI failures  such as https://cirrus-ci.com/task/5511952184115200?logs=ci#L4024, I think it does matter:
`peer_disconnect`  doesn't wait until the node has completed the disconnection. So there is a race between setting up the next connection (next  `addconnection` RPC), and if the old one hasn't been removed and is identical to the new one (because we didn't increment the ID), CConnman::OpenNetworkConnection just [returns](https://github.com/bitcoin/bitcoin/blob/5274f324375fd31cf8507531fbc612765d03092f/src/net.cpp#L1976) without establishing a connection.

Reply:
Good point! I think you're right, I've been looking at it for a bit and haven't found a better explanation. Please have this requirement documented in the PR you gonna open :)

Reply:
Done in #26448